index.js

```diff
@@ -52,9 +53,22 @@ function doRequest(method, url, options, callback) {
     var body = options.body ? options.body : new Buffer(0);
     if (typeof body === 'string') body = new Buffer(body);
     assert(Buffer.isBuffer(body), 'body should be a Buffer or a String');
-    if (!headers.has('Content-Length')) {
+    if (!headers.has('Content-Length') && !options.form) {
       headers.set('Content-Length', body.length);
     }
+    if (options.form) {
+      var form = new FormData();
```

Comment:
Please don't rely on the odd scoping rules of `var`, imagine it is `let`.  Please declare `var form = null;` outside of the if block, then use `form = new FormData();` inside the if block.